gnosis/safe-react
# Feature/2397 display guard information in settings

## What it solves
Resolves #2397 

## How this PR fixes it

We display a table containing the transaction guard information with a button to delete the transaction guard

Creating the PR as draft since i'm not sure how ModulePair works and how to set a transaction guard

## How to test it

:warning: Adding a guard could lock the safe if this isn't done correctly. Usually you would deploy a guard for yourself and have some handling mechanism. For the Guard we were provided for testing purposes, is owner by shared QA testing account, so can be handled by this account. This guard will prevent any operation within a Safe unless explicitly enabled in the Guard.

:warning: In order to be able to remove the Guard you need to enable in the guard the possibility for the safe to execute transactions to itself.

1.- Make sure you have access to TestOwner key or check with Liliy
2.- Use the transaction builder to create a transaction using `setGuard` method of your safe configuring this safe Guard `0x4f8a82d73729A33E0165aDeF3450A7F85f007528`
3.- After correctly enabling the guard test that now its showed under `Settings` -> `Advanced`
4.- Check that currently you can't execute any transactions with your safe. Try a send funds or any settings change and you won't be able to do anything
5.- When you are ready you can go [here](https://rinkeby.etherscan.io/address/0x4f8a82d73729A33E0165aDeF3450A7F85f007528#writeContract) and add your safeAddress to "allowTarget" function and submit the transaction so it's stored in the Guard
6.- BONUS try to enable and disable some functions at different points in the guard and check that the safe behaves as expected. To remove the guard make sure that "allowTarget" contains your safeAddress and you will be able to remove the Guard
